src/main/java/org/apache/commons/configuration2/interpol/StringLookupCatchExceptionAdapter.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.configuration2.interpol;
+
+import org.apache.commons.text.lookup.StringLookup;
+
+/**
+ * Extension of StringLookupAdapter, the difference is return null if catching exception during lookup. 
+ *
+ * @since 2.8.0
+ */
+class StringLookupCatchExceptionAdapter extends StringLookupAdapter
```

Comment:
Maybe `StringLookupAdapter` could have a parameter like `suppress=false` that's disabled by default? This class is not public, so either way is fine, we can improve on it later IMO.

Reply:
One issue I see here is that if we suppress exceptions in the adapter, then legitimate configuration errors will not be propagated. For example, say I have a property like `current date= ${date:not-a-format}`. With the changes here, this would interpolate without error to the string `current date= ${date:not-a-format}` instead of throwing an exception about the invalid format string, as is the current behavior.